module-2/lab-probability-distribution/your_code/main.ipynb

```diff
@@ -12,15 +12,18 @@
   },
```

Comment:
- Sometimes it's useful to seed the random number generator. Seeding a pseudo-random number generator gives it its first "previous" value. Each seed value will correspond to a sequence of generated values for a given random number generator. That is, if you provide the same seed twice, you get the same sequence of numbers twice. https://machinelearningmastery.com/how-to-generate-random-numbers-in-python/
-Challenge 1 - Generate and Plot Normal Distributions: Try to do this challenge in a single line for all sample sizes. Figure it out how to transform the procedural you have done into a more structured chunk of code. 
- Normality tests are quite sensitive to sample size thus for a sample size below a couple of thousands it should be a better idea to evaluate normality through a plot, in particular a quantile-quantile plot (QQ plot). A Q-Q plot is a scatterplot created by plotting two sets of quantiles against one another. It plots the quantiles of our sample vs the theoretical quantiles of the distribution type we want to test (this case a normal distribution) How to do this in Python? https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.probplot.html.
- There are 2 other normality tests (additional to D'Agostino k^2 test, normaltest in Python) that are very used in to test normality of our data. These are Kolmogorov-Smirnof and Shapiro-Wilk tests. I encourage you to deepen on these type of tests. In case you are interested, I can provide you some useful resources. 
-  "si esto se cumple significa que la distribucion es normal si no, no es mormal" ,  cuando corremos un test de contraste de hipótesis, si los estadísticos de contraste están dentro del rango que le hemos establecido decimos que "no podemos rechazar la hipótesis nula". Esto lo decimos porque a pesar de un resultado positivo del estadístico de contraste podría no ser normal. Esto lo ves claramente en el primer histograma tamaño de la muestra 5, el p-value te dice que no puedes rechazar la hipótesis nula de normalidad y, sin embargo, el histograma muestra algo que no tiene nada que ver con una normal. Aquí podríamos graficar un gráfico de quantiles y, probablemente, veríamos, al igual que en el histograma, que los datos no vienen de una distribución normal. 
- When you do this for comparing, I would suggest to let python to set the bins automatically otherwise your are not comparing same bining which it's important from a visual point of view. 
```Python
fig,axes= plt.subplots(1,4,figsize=(16,6))
sns.distplot(sample10, bins=5, kde=True,ax=axes[0])
sns.distplot(sample50, bins=20, kde=True,ax=axes[1])
sns.distplot(sample500, bins=50, kde=True,ax=axes[2])
sns.distplot(sample5000, bins=100, kde=True,ax=axes[3])
```
- Interpretation of the CLT: take a look on these 2 articles (the most clear explanations I've found). You can replicate in Python some of the examples they include. 
https://towardsdatascience.com/understanding-the-central-limit-theorem-642473c63ad8. 
- Normal distribution can be used as an approximation of Binomial and Poisson distributions under certain circumstances. Generally speaking: 
      Binomial:  X ~ B(n, p) and if n is large and/or p is close to ½, then X is approximately N(np, npq)
      Poisson: X ~ Po(l) then for large values of l, X ~ N(l, l) approximately.
However, we must be careful because binomial and poisson distributions are discrete while Normal distribution is continuous. 
- Poisson: expresses the probability of a given number of events occurring in a fixed interval of time or space if these events occur with a known constant rate and independently of the time since the last event.
- Good one!! 
```Python
fig, ax = plt.subplots(1, 1)
mu=300
x = np.arange(0,1000)
ax.plot(x, sci.poisson.pmf(x, mu), 'bo', ms=8, label='poisson pmf')
```
- Try to set x and y labels as well as the title when plotting. It's a very good practice.